File: Methods-Python/computeCanberraDistance.py
def computeCanberraDistance(a, b):

    sum = 0

    for i in range(0, len(a)):
        num = abs(a[i] - b[i])
        denom = abs(a[i]) + abs(b[i])
        if num == 0 or denom == 0:
            continue
        sum += num/denom

    return sum

File: Methods-C++/computeCanberraDistance.cpp
#include <iostream>

double computeCanberraDistance(double a[], double b[], int n) {
    double sum = 0;
    for (int i = 0; i < n; i++) {
        double num = abs(a[i] - b[i]);
        double denom = abs(a[i]) + abs(b[i]);
        sum += num == 0.0 && denom == 0.0 ? 0.0 : num / denom;
    }
        return sum;
    }

